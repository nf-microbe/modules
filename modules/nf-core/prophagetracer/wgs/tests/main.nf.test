nextflow_process {

    name "Test Process PROPHAGETRACER_WGS"
    script "../main.nf"
    process "PROPHAGETRACER_WGS"

    test("fasta.gz - bam") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/CarsonJM/HoffLab-phageannotator/prophage_tracer_setup/NC_013967_1_genome.fasta.gz", checkIfExists: true) ]
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    [ file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/CarsonJM/HoffLab-phageannotator/prophage_tracer_setup/test.rmdup.bam", checkIfExists: true) ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - bam - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [ file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/CarsonJM/HoffLab-phageannotator/prophage_tracer_setup/NC_013967_1_genome.fasta.gz", checkIfExists: true) ]
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    [ file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/CarsonJM/HoffLab-phageannotator/prophage_tracer_setup/test.rmdup.bam", checkIfExists: true) ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}