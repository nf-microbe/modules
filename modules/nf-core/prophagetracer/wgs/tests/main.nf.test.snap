{
    "fasta.gz - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ],
                "prophage": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-16T11:27:20.147806848"
    },
    "fasta.gz - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d61244e2b46c77c9fd56259f1e953099"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ],
                "prophage": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d61244e2b46c77c9fd56259f1e953099"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-16T11:27:10.134297791"
    }
}
